From 4ac9c58d275311f38836e6aed5b14709c8cc6b3a Mon Sep 17 00:00:00 2001
From: ctxnop <ctxnop@gmail.com>
Date: Mon, 15 May 2023 19:30:16 +0200
Subject: [PATCH] Fix #1558: Don't assume inclusion of cstdint

With last versions of gcc, some headers don't include cstdint anymore,
but some sources assume that it is.

Signed-off-by: ctxnop <ctxnop@gmail.com>
---
 libdnf/conf/ConfigMain.hpp       | 1 +
 libdnf/conf/ConfigRepo.hpp       | 1 +
 libdnf/conf/OptionNumber.hpp     | 1 +
 libdnf/conf/OptionSeconds.hpp    | 2 ++
 libdnf/utils/sqlite3/Sqlite3.hpp | 1 +
 5 files changed, 6 insertions(+)

diff --git a/libdnf/conf/ConfigMain.hpp b/libdnf/conf/ConfigMain.hpp
index 41368b76..ae032898 100644
--- a/libdnf/conf/ConfigMain.hpp
+++ b/libdnf/conf/ConfigMain.hpp
@@ -32,6 +32,7 @@
 #include "OptionString.hpp"
 #include "OptionStringList.hpp"
 
+#include <cstdint>
 #include <memory>
 
 namespace libdnf {
diff --git a/libdnf/conf/ConfigRepo.hpp b/libdnf/conf/ConfigRepo.hpp
index 414de9c4..132f22da 100644
--- a/libdnf/conf/ConfigRepo.hpp
+++ b/libdnf/conf/ConfigRepo.hpp
@@ -26,6 +26,7 @@
 #include "ConfigMain.hpp"
 #include "OptionChild.hpp"
 
+#include <cstdint>
 #include <memory>
 
 namespace libdnf {
diff --git a/libdnf/conf/OptionNumber.hpp b/libdnf/conf/OptionNumber.hpp
index f7a7b3d6..a3a4dea6 100644
--- a/libdnf/conf/OptionNumber.hpp
+++ b/libdnf/conf/OptionNumber.hpp
@@ -25,6 +25,7 @@
 
 #include "Option.hpp"
 
+#include <cstdint>
 #include <functional>
 
 namespace libdnf {
diff --git a/libdnf/conf/OptionSeconds.hpp b/libdnf/conf/OptionSeconds.hpp
index dc714b23..a80a973f 100644
--- a/libdnf/conf/OptionSeconds.hpp
+++ b/libdnf/conf/OptionSeconds.hpp
@@ -25,6 +25,8 @@
 
 #include "OptionNumber.hpp"
 
+#include <cstdint>
+
 namespace libdnf {
 
 /**
diff --git a/libdnf/utils/sqlite3/Sqlite3.hpp b/libdnf/utils/sqlite3/Sqlite3.hpp
index 3a7da23c..81a731aa 100644
--- a/libdnf/utils/sqlite3/Sqlite3.hpp
+++ b/libdnf/utils/sqlite3/Sqlite3.hpp
@@ -32,6 +32,7 @@
 #include <stdexcept>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 class SQLite3 {
 public:
